pkg/kubelet/cm/dra/types.go

```diff
@@ -17,43 +17,9 @@ limitations under the License.
 package dra
 
 import (
-	"context"
-
-	v1 "k8s.io/api/core/v1"
-	"k8s.io/apimachinery/pkg/types"
-	"k8s.io/kubernetes/pkg/kubelet/config"
 	kubecontainer "k8s.io/kubernetes/pkg/kubelet/container"
-	"k8s.io/kubernetes/pkg/kubelet/pluginmanager/cache"
 )
 
-// Manager manages all the DRA resource plugins running on a node.
-type Manager interface {
-	// GetWatcherHandler returns the plugin handler for the DRA.
-	GetWatcherHandler() cache.PluginHandler
-
-	// Start starts the reconcile loop of the manager.
-	// This will ensure that all claims are unprepared even if pods get deleted unexpectedly.
-	Start(ctx context.Context, activePods ActivePodsFunc, getNode GetNodeFunc, sourcesReady config.SourcesReady) error
-
-	// PrepareResources prepares resources for a pod.
-	// It communicates with the DRA resource plugin to prepare resources.
-	PrepareResources(ctx context.Context, pod *v1.Pod) error
-
-	// UnprepareResources calls NodeUnprepareResource GRPC from DRA plugin to unprepare pod resources
-	UnprepareResources(ctx context.Context, pod *v1.Pod) error
-
-	// GetResources gets a ContainerInfo object from the claimInfo cache.
-	// This information is used by the caller to update a container config.
-	GetResources(pod *v1.Pod, container *v1.Container) (*ContainerInfo, error)
-
-	// PodMightNeedToUnprepareResources returns true if the pod with the given UID
-	// might need to unprepare resources.
-	PodMightNeedToUnprepareResources(UID types.UID) bool
-
-	// GetContainerClaimInfos gets Container ClaimInfo objects
-	GetContainerClaimInfos(pod *v1.Pod, container *v1.Container) ([]*ClaimInfo, error)
-}
```

Comment:
This was added to simplify mocking in the unit tests, but seems to not used. I agree, it can be removed, unless we come up with unit tests that want to mock Manager, e.g. if they don't want to run gRPC server.

Reply:
+1